services/data/postgres_async_db.py

```diff
@@ -428,12 +442,52 @@ async def create_trigger_if_missing(db: _AsyncPostgresDB, table_name, trigger_na
                 cur.close()
 
     @staticmethod
-    async def setup_trigger_notify(db: _AsyncPostgresDB, table_name, keys: List[str] = None, schema=DB_SCHEMA_NAME):
+    async def cleanup_triggers(db: _AsyncPostgresDB, table_name):
+        "Cleans up old versions of table triggers"
+        with (await db.pool.cursor()) as cur:
+            try:
+                await cur.execute(
+                    """
+                    SELECT DISTINCT trigger_name, trigger_schema
+                    FROM information_schema.triggers
+                    WHERE event_object_table = %s
+                    """,
+                    [table_name]
+                )
+                results = await cur.fetchall()
+
+                triggers_to_cleanup = [
+                    (res[0], res[1]) for res in results
+                    if res[0].startswith(TRIGGER_NAME_PREFIX) and TRIGGER_VERSION not in res[0]
+                ]
+                if triggers_to_cleanup:
+                    logging.getLogger("TriggerSetup").info("Cleaning up old triggers: %s" % triggers_to_cleanup)
+                    commands = []
+                    for trigger_name, schema in triggers_to_cleanup:
+                        commands += [
+                            (f"DROP TRIGGER IF EXISTS {trigger_name} ON {table_name}"),
+                            (f"DROP FUNCTION IF EXISTS {schema}.{trigger_name}")
```

Comment:
@romain-intel This is the only major difference to the setup in the previous PR

as suggested, I now check for the known prefix for ui-triggers, as well as look for the expected version before deciding if a trigger needs to be cleaned up or not.

I also added dropping the actual trigger function, as previously I was only uncoupling the function from the table.

Reply:
whether the checks are computationally expensive or not is not relevant here, as the code runs only during server startup when triggers are setup.